core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -102,37 +106,14 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
       deleteFiles(io, manifestsToDelete);
     }
 
-    Tasks.foreach(Iterables.transform(manifestsToDelete, ManifestFile::path))
-        .executeWith(ThreadPools.getWorkerPool())
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure((manifest, exc) -> LOG.warn("Delete failed for manifest: {}", manifest, exc))
-        .run(io::deleteFile);
-
-    Tasks.foreach(manifestListsToDelete)
-        .executeWith(ThreadPools.getWorkerPool())
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure((list, exc) -> LOG.warn("Delete failed for manifest list: {}", list, exc))
-        .run(io::deleteFile);
-
-    Tasks.foreach(
-            Iterables.transform(metadata.previousFiles(), TableMetadata.MetadataLogEntry::file))
-        .executeWith(ThreadPools.getWorkerPool())
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure(
-            (metadataFile, exc) ->
-                LOG.warn("Delete failed for previous metadata file: {}", metadataFile, exc))
-        .run(io::deleteFile);
-
-    Tasks.foreach(metadata.metadataFileLocation())
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure(
-            (metadataFile, exc) ->
-                LOG.warn("Delete failed for metadata file: {}", metadataFile, exc))
-        .run(io::deleteFile);
+    deleteFiles(io, Iterables.transform(manifestsToDelete, ManifestFile::path), "manifest", true);
+    deleteFiles(io, manifestListsToDelete, "manifest list", true);
+    deleteFiles(
+        io,
+        Iterables.transform(metadata.previousFiles(), TableMetadata.MetadataLogEntry::file),
+        "previous metadata",
+        true);
+    deleteFiles(io, ImmutableList.of(metadata.metadataFileLocation()), "metadata", true);
```

Comment:
nit: Just `deleteFile` as we have only one file?

Reply:
Good point, updated!